src/nor_flash.rs

```diff
@@ -379,3 +379,350 @@ where
 		Ok(())
 	}
 }
+
+/// Simple RAM-backed flash storage implementation for tests
+#[derive(Clone, Copy, Debug)]
+pub struct MockFlash<
```

Comment:
Could you move this inside the `mod tests` module?

Reply:
The idea is export this struct to use for testing dependent crates.